test/Mono.Linker.Tests.Cases/Reflection/ExpressionNewType.cs

```diff
@@ -20,8 +42,38 @@ class T1
 			[Kept]
 			internal T1 ()
 			{
-
 			}
 		}
+
+		class T2
+		{
+		}
+
+		[Kept]
+		static void Branch_SystemTypeValueNode ()
+		{
+			var expr = Expression.New (typeof (T1));
+		}
+
+		[Kept]
+		static void Branch_NullValueNode ()
+		{
```

Comment:
Similar comment as above - add a case where there are two potential values for the type - you kind of have it here by having null and a type, but maybe another one with two existing types (again, they all have to be only used exactly once for the validation to work).